src/DataWrangling/ECCO/ECCO_metadata.jl

```diff
@@ -150,6 +152,10 @@ function download_dataset(metadata::ECCOMetadata)
     password = get(ENV, "ECCO_PASSWORD", nothing)
     dir = metadata.dir
 
+    if !isdir(dir)
+        mkpath(dir)
+    end
```

Comment:
Should we make a directory automatically? I would probably prefer making sure that the users make sure the directory exists rather than automatically touching the filesystem.

Reply:
i.e., just spit an informative error if the directory does not exist

Reply:
I agree. Also using `dir` should be discouraged.

Reply:
I see your point!

I was doing that because I thought we'd better have the ECCO2 and ECCO4 data in different directories. Also different directories for ECCO2 monthly vs daily. 

But perhaps it's better to have 3 different scratch dirs?

https://github.com/CliMA/ClimaOcean.jl/blob/58065e8516db6272699d6530f34b095886c84c11/src/DataWrangling/ECCO/ECCO.jl#L31-L34

Reply:
I think if we want to look at the data, then we need utilities to copy or create symlinks to it. I think it's also not proper to go to the scratch directory. In summary:

1. Always download data to scratch for simulations. This ensures that data is not downloaded repeatedly.
2. If you would like to look at data in a way that is not accessible through ClimaOcean utilities, then copy from scratch to a local location. (Or, implement the feature that is missing in ClimaOcean.)

Reply:
I agree!

But I'm a bit confused. I don't wanna look at any data. Just want to ensure that data is not being overwritten. E.g. both daily and monthly ECCO2 SSH has the same filename (see https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/daily/SSH and https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/monthly/SSH). Are we sure that if monthly is downloaded after the daily one doesn't overwrite the other?

I was just suggesting above that the cleanest-safest way is that each dataset has its own scratch directory. Perhaps I didn't explain myself very well?

Reply:
oh well for sure if they have the same name then I think you are right, different directories are needed. I didn't realize that part!

Reply:
I didn't understand because you said "we'd better have them in different directories" rather than "we need them in different directories [because the filenames overlap]"

Reply:
oh no worries

so what do you think:

- 3 different scratch directories?
or
- 1 scratch directory for ECCO and subdirectories for ECCO2, ECCO4 / daily, monthly?

Reply:
> Should we make a directory automatically? I would probably prefer having the users make sure the directory exists rather than automatically touching the filesystem.

see [2205be1](https://github.com/CliMA/ClimaOcean.jl/pull/479/commits/2205be174eeec348010b396dc57c1de3d8f7eb33) does the former